parsedatetime/__init__.py

```diff
@@ -863,7 +898,166 @@ def _quantityToReal(self, quantity):
+                        million_multiplier = self.number_formation(clean_numbers[0:million_index])
+                    total_sum += million_multiplier * 1000000
+
+                if thousand_index > -1:
+                    if million_index > -1:
+                        thousand_multiplier = self.number_formation(clean_numbers[million_index + 1:thousand_index])
+                    elif billion_index > -1 and million_index == -1:
+                        thousand_multiplier = self.number_formation(clean_numbers[billion_index + 1:thousand_index])
+                    else:
+                        thousand_multiplier = self.number_formation(clean_numbers[0:thousand_index])
+                    total_sum += thousand_multiplier * 1000
+
+                if thousand_index > -1 and thousand_index != len(clean_numbers) - 1:
+                    hundreds = self.number_formation(clean_numbers[thousand_index + 1:])
+                elif million_index > -1 and million_index != len(clean_numbers) - 1:
+                    hundreds = self.number_formation(clean_numbers[million_index + 1:])
+                elif billion_index > -1 and billion_index != len(clean_numbers) - 1:
+                    hundreds = self.number_formation(clean_numbers[billion_index + 1:])
+                elif thousand_index == -1 and million_index == -1 and billion_index == -1:
+                    hundreds = self.number_formation(clean_numbers)
+                else:
+                    hundreds = 0
+                total_sum += hundreds
+
+        # adding decimal part to total_sum (if exists)
+        if len(clean_decimal_numbers) > 0:
+            decimal_sum = get_decimal_sum(clean_decimal_numbers)
+            total_sum += decimal_sum
+
+        return float(total_sum)
+
+    def replaceNumber(self, parseStr, tense=None):
+        if tense is None or tense == 0:
+            tense = 1
+        if self.ptc.CRE_NUM.search(parseStr) is not None:
+            chunk1 = chunk2 = num = ''
+            strWords = parseStr.split()
+            flag = 0
+            for i in strWords:
+                m = self.ptc.CRE_NUM.search(i + ' ')
+                if m is not None and\
+                    self.ptc.CRE_MONTH.search(i) is None and \
+                    self.ptc.CRE_WEEKDAY.search(i) is None:
+                    flag = 1
+                    num = num + i + ' '
+                elif flag == 1:
+                    chunk2 = chunk2 + ' ' + i
+                elif flag == 0:
+                    chunk1 = chunk1 + ' ' + i
+            if flag:
+                number = self.word_to_num(parseStr)
+                number = number * tense
+                return (chunk1.strip() + ' ' + str(int(number)) + ' ' + chunk2.strip().strip()).strip()
+            else:
+                return parseStr
+        elif tense == -1:
+            return self.addContext(parseStr, tense)
+        else:
+            return parseStr
+
+    def addContext(self, parseStr, tense):
```

Comment:
What does Context signify?

Reply:
i changed the function name to addTense, to a phrase like '4 days' and there is tense = -1, then the phrase gets changed to '-4.0 days' and gets parsed